src/proxy-ui-api/frontend/tests/e2e/pages/ssMainPage.js

```diff
@@ -15,35 +15,234 @@ var navigateCommands = {
   openSettingsTab: function() {
     this.click('@settingsTab');
     return this;
+  },
+  logout: function() {
+    this.click('@userMenuButton');
+    this.pause(1000);
+    this.click('@userMenuitemLogout');
+    return this;
+  },
+  acceptLogout: function() {
+    this.click('@logoutOKButton');
+    return this;
+  }
+};
+
+var clientTabCommands = {
+  clickNameHeader: function() {
+    this.click('@listNameHeader');
+    return this;
+  },
+  clickIDHeader: function() {
+    this.click('@listIDHeader');
+    return this;
+  },
+  clickStatusHeader: function() {
+    this.click('@listStatusHeader');
+    return this;
+  },
+  openTestGov: function() {
+    this.click('@testGovListItem');
+    return this;
+  },
+  openTestService: function() {
+    this.click('@testServiceListItem');
+    return this;
+  }
+};
+
+var clientInfoCommands = {
+  openDetailsTab: function() {
+    this.click('@detailsTab');
+    return this;
+  },
+  openServiceClientsTab: function() {
+    this.click('@serviceClientsTab');
+    return this;
+  },
+  openServicesTab: function() {
+    this.click('@servicesTab');
+    return this;
+  },
+  openInternalServersTab: function() {
+    this.click('@internalServersTab');
+    return this;
+  },
+  openLocalGroupsTab: function() {
+    this.click('@localGroupsTab');
+    return this;
+  }
+};
+
+var clientDetailsCommands = {
+  openSignCertificateInfo: function() {
+    this.click('@clientSignCertificate');
+    return this;
+  }
+};
+
+var certificatePopupCommands = {
+  close: function() {
+    this.click('@certificateInfoCloseButton');
+    return this;
+  }
+};
+
+
+var clientLocalGroupsCommands = {
+  openAddLocalGroupDialog: function() {
+    this.click('@addGroupButton');
+    return this;
+  },
+  filterBy: function(filter) {
+    this.clearValue2('@filterInput');
+    this.setValue('@filterInput', filter);
+    return this;
+  },
+  openAddDialog: function() {
+    this.click('@addGroupButton');
+    return this;
+  },
+  confirmAddDialog: function() {
+    this.click('@confirmAddButton');
+    return this;
+  },
+  cancelAddDialog: function() {
+    this.click('@cancelAddButton');
+    return this;
+  },
+  enterCode: function(code) {
+    this.clearValue2('@groupCode');
+    this.setValue('@groupCode', code);
+    return this;
+  },
+  enterDescription: function(description) {
+    this.clearValue2('@groupDescription');
+    this.setValue('@groupDescription', description);
+    return this;
   }
 };
 
-var clientTabCommands = {};
 
 module.exports = {
   url: process.env.VUE_DEV_SERVER_URL,
   commands: [navigateCommands],
   elements: {
     clientsTab: { 
-      selector: '//div[contains(@class, "v-tabs__bar")]//a[text()="clients"]', 
+      selector: '//div[contains(@class, "v-tabs-bar__content")]//a[text()="Clients"]', 
```

Comment:
Depending on generated class-names can crash when the library generating the html is updated. This relates to the comment below. There seems to be several issues like this so I won't create more comments about those. 

No need to act on this as the situation may be getting better considering those test-data attributes?